Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -282,7 +274,21 @@ namespace MultiplayerSample
 
         RPC_BallExplosion(m_hitEvent);
 
-        Multiplayer::GetNetworkEntityManager()->MarkForRemoval(GetNetBindComponent()->GetEntityHandle());
+        // Wait 5 seconds before cleaning up the entity so that the explosion effect has a chance to play out
+        // Capture just the netEntityId in case we have a level change or some other operation that clears out entities before our lamda triggers
+        const Multiplayer::NetEntityId netEntityId = GetNetEntityId();
+        AZ::Interface<AZ::IEventScheduler>::Get()->AddCallback([netEntityId]
+            {
+                // Fetch the entity handle, ensure its still valid
+                const Multiplayer::ConstNetworkEntityHandle entityHandle = Multiplayer::GetNetworkEntityManager()->GetEntity(netEntityId);
+                if (entityHandle.Exists())
+                {
+                    Multiplayer::GetNetworkEntityManager()->MarkForRemoval(entityHandle);
+                }
+            },
+            AZ::Name("Cleanup"),
+            AZ::TimeMs{ 5000 }
```

Comment:
Should the time be exposed out as a tunable property?

Reply:
Yup, good call, done.